Apps/FlightSoftware/Watchdog2/src/i2c.c

```diff
@@ -0,0 +1,318 @@
+    case USCI_I2C_UCALIFG:   break;         // Vector 2: ALIFG
+    case USCI_I2C_UCNACKIFG:                // Vector 4: NACKIFG
+      break;
+    case USCI_I2C_UCSTTIFG:  break;         // Vector 6: STTIFG
+    case USCI_I2C_UCSTPIFG:  break;         // Vector 8: STPIFG
+    case USCI_I2C_UCRXIFG3:  break;         // Vector 10: RXIFG3
+    case USCI_I2C_UCTXIFG3:  break;         // Vector 12: TXIFG3
+    case USCI_I2C_UCRXIFG2:  break;         // Vector 14: RXIFG2
+    case USCI_I2C_UCTXIFG2:  break;         // Vector 16: TXIFG2
+    case USCI_I2C_UCRXIFG1:  break;         // Vector 18: RXIFG1
+    case USCI_I2C_UCTXIFG1:  break;         // Vector 20: TXIFG1
+    case USCI_I2C_UCRXIFG0:                 // Vector 22: RXIFG0
+        rx_val = UCB0RXBUF;
+        if (RXByteCtr)
+        {
+          ReceiveBuffer[ReceiveIndex++] = rx_val;
+          RXByteCtr--;
+        }
+
+        if (RXByteCtr == 1)
+        {
+          UCB0CTLW0 |= UCTXSTP;
+        }
+        else if (RXByteCtr == 0)
+        {
+          UCB0IE &= ~UCRXIE;
+          MasterMode = IDLE_MODE;
+//          __bic_SR_register_on_exit(CPUOFF);      // Exit LPM0
+        }
+        break;
+    case USCI_I2C_UCTXIFG0:                 // Vector 24: TXIFG0
+        switch (MasterMode)
+        {
+          case TX_REG_ADDRESS_MODE:
+              UCB0TXBUF = TransmitRegAddr;
+              if (RXByteCtr)
+                  MasterMode = SWITCH_TO_RX_MODE;   // Need to start receiving now
+              else
+                  MasterMode = TX_DATA_MODE;        // Continue to transmision with the data in Transmit Buffer
+              break;
+
+          case SWITCH_TO_RX_MODE:
+              UCB0IE |= UCRXIE;              // Enable RX interrupt
+              UCB0IE &= ~UCTXIE;             // Disable TX interrupt
+              UCB0CTLW0 &= ~UCTR;            // Switch to receiver
+              MasterMode = RX_DATA_MODE;    // State state is to receive data
+              UCB0CTLW0 |= UCTXSTT;          // Send repeated start
+              if (RXByteCtr == 1)
+              {
+                  //Must send stop since this is the N-1 byte
+                  while((UCB0CTLW0 & UCTXSTT));
+                  UCB0CTLW0 |= UCTXSTP;      // Send stop condition
+              }
+              break;
+
+          case TX_DATA_MODE:
+              if (TXByteCtr)
+              {
+                  UCB0TXBUF = TransmitBuffer[TransmitIndex++];
+                  TXByteCtr--;
+              }
+              else
+              {
+                  //Done with transmission
+                  UCB0CTLW0 |= UCTXSTP;     // Send stop condition
+                  MasterMode = IDLE_MODE;
+                  UCB0IE &= ~UCTXIE;                       // disable TX interrupt
+//                  __bic_SR_register_on_exit(CPUOFF);      // Exit LPM0
+              }
+              break;
+
+          default:
+              __no_operation();
```

Comment:
What's the point of this?

Reply:
Not sure, artefact of the i2c template I used. Removed.